src/python/pants/base/specs_parser.py

```diff
@@ -44,8 +44,17 @@ class SpecsParser:
     class BadSpecError(Exception):
         """Indicates an unparseable command line selector."""
 
-    def __init__(self, root_dir: str | None = None) -> None:
+    def __init__(self, root_dir: str | None = None, work_dir: str | None = None) -> None:
```

Comment:
```suggestion
    def __init__(self, *, root_dir: str | None = None, work_dir: str | None = None) -> None:
```

Since these args are both "dirs" and not distinguished by type, let's at least force the caller to name them, for clarity to the reader, and to avoid bugs incurred by swapping them accidentally.